rfcs/text/????-hook-service.md

```diff
@@ -0,0 +1,134 @@
+- Feature Name: `hook-service`
+- Start Date: (fill me in with today's date, 2022-07-12)
+- RFC PR: [blockprotocol/blockprotocol#430](https://github.com/blockprotocol/blockprotocol/pull/430)
+- Block Protocol Discussion: N/A
+
+# Summary
+
+[summary]: #summary
+
+This service standardises the communication between blocks and the embedding application necessary to enable embedding applications to 'take over' the rendering of parts of a block to extend functionality, while ensuring block can remain uncoupled from the embedding application, even where these are implemented by the same organisation.
+
+In order to enable direct manipulation, this service is likely to be difficult or impossible to implement in Embedding Applications which use common ‘sandboxing’ techniques including `iframe`s.
+
+# Motivation
+
+[motivation]: #motivation
+
+It is not uncommon for embedding applications to want to provide a consistent 'look and feel' across distinct blocks which implement similar features – i.e, rich text or image editing. Additionally, the implementation of these features may require integrating with systems bespoke to the embedding application. We don't want blocks to have to 'know' about the implementation of this functionality, or necessarily the shape of the data involved in providing this functionality. Therefore, the embedding application needs to be responsible for providing this functionality, but this depends on co-operation by block authors. 
+
+# Guide-level explanation
+
+[guide-level-explanation]: #guide-level-explanation
+
+A 'hook' is an injection point blocks provide to embedding applications to allow them to optionally render or modify already rendered views associated with a path to data provided by the graph service's `blockEntity` property. 
+
+The 'optionally' here is key. Blocks cannot require embedding applications to implement a handler for any specific 'hook'. Therefore, each 'hook' must specify a fallback rendering strategy which embedding applications implementing the hook service can use instead of implementing the hook themselves.   
```

Comment:
```suggestion
**Note: the word 'optionally' here is key.** Blocks cannot today _require_ embedding applications to implement a handler for any specific 'hook'. Therefore, each 'hook' must specify a fallback rendering strategy which embedding applications implementing the hook service can use instead of implementing the hook themselves.
```

Not sure if we mean 'must' here or 'should'. There may be instances where a block could continue to function without an interface/service being passed in to a single given hook, depending on how these end up getting used.

More broadly we should capitalize [RFC 2119](https://datatracker.ietf.org/doc/html/rfc2119) terms for clarity in our own RFCs and Service Specifications, as we do in the [Core Specification](https://blockprotocol.org/docs/spec/core-specification).